src/cata_tiles.cpp

```diff
@@ -825,36 +841,60 @@ bool cata_tiles::draw_tile_at(tile_type *tile, int x, int y, int rota)
 
     int ret = 0;
     // blit foreground based on rotation
-    if (rota == 0) {
-        if (fg >= 0 && static_cast<size_t>( fg ) < tile_values.size()) {
-            SDL_Texture *fg_tex = tile_values[fg];
-            ret = SDL_RenderCopyEx( renderer, fg_tex, NULL, &destination, 0, NULL, SDL_FLIP_NONE );
-        }
+    int rotate_sprite, sprite_num;
+    if ( fg.empty() ) { 
+        // render nothing
     } else {
-        if (fg >= 0 && static_cast<size_t>( fg ) < tile_values.size()) {
-            SDL_Texture *fg_tex = tile_values[fg];
+        if ( fg.size() == 1 ) {
+            // just one tile, apply SDL sprite rotation
+            rotate_sprite = true;
+            sprite_num = 0;
+        } else {
+            // multiple rotated tiles defined, don't apply sprite rotation after picking one
+            rotate_sprite = false;
+            // two tiles, tile 0 is N/S, tile 1 is E/W
+            // four tiles, 0=N, 1=E, 2=S, 3=W
+            // extending this to more than 4 rotated tiles will require changing rota to degrees
+            sprite_num = rota % fg.size();
+        }
 
-            if(rota == 1) {
-#if (defined _WIN32 || defined WINDOWS)
-                destination.y -= 1;
-#endif
-                ret = SDL_RenderCopyEx( renderer, fg_tex, NULL, &destination,
-                    -90, NULL, SDL_FLIP_NONE );
-            } else if(rota == 2) {
-                //flip rather then rotate here
-                ret = SDL_RenderCopyEx( renderer, fg_tex, NULL, &destination,
-                    0, NULL, static_cast<SDL_RendererFlip>( SDL_FLIP_HORIZONTAL | SDL_FLIP_VERTICAL ) );
-            } else { //rota == 3
-#if (defined _WIN32 || defined WINDOWS)
-                destination.x -= 1;
-#endif
+        if ( static_cast<size_t>( fg[0] ) < tile_values.size() ) {
```

Comment:
`fg` and `bg` needed to be `int` to store the -1 value (=no tile), now this is not needed anymore (no tile is indicated by an empty vector instead). At this point it might be useful to change the entry type of that vector to `size_t` to avoid this repeated cast.

It might also be useful to remove invalid entries (e.g. those that have a value >= tile_vales.size()) right after loading the whole tileset. That would allow you to remove this check (`draw_tile_at` is performance critical).